core/src/main/java/nl/nn/adapterframework/http/HttpSender.java

```diff
@@ -243,33 +281,50 @@ protected HttpRequestBase getMethod(URI uri, String message, ParameterValueList
 				return method;
 			} else if (getMethodType().equals("POST")) {
 				HttpPost method = new HttpPost(relativePath.toString());
-				if (parameters!=null) {
-					StringBuffer msg = new StringBuffer(message);
-					appendParameters(true,msg,parameters);
-					if (StringUtils.isEmpty(message) && msg.length()>1) {
-						message=msg.substring(1);
-					} else {
-						message=msg.toString();
+
+				HttpEntity entity;
+				if(postType.equals(PostType.RAW)) {
+					String messageString = message.asString();
+					if (parameters!=null) {
+						StringBuffer msg = new StringBuffer(messageString);
+						appendParameters(true,msg,parameters);
+						if (StringUtils.isEmpty(messageString) && msg.length()>1) {
+							messageString=msg.substring(1);
+						} else {
+							messageString=msg.toString();
+						}
 					}
+					entity = new ByteArrayEntity(messageString.getBytes(getCharSet()), getFullContentType());
+				} else if(postType.equals(PostType.BINARY)) {
+					entity = new ByteArrayEntity(message.asByteArray(getCharSet()), getFullContentType());
```

Comment:
Hier zou je in plaats van een ByteArrayEntity ook de content pas kunnen streamen als het nodig is.